### DISCUSSION TITLE
WSHttpBinding Message Security Implementation with Windows Identity

### DISCUSSION THREAD

**Discussion by DublinDel** [CoreWCF/CoreWCF#334] [Q&A]
Hi there

(If this is the wrong forum for Q+A and you are now using the  [  https://gitter.im/CoreWCF/CoreWCF](url) site then I can take it over there.)

This might be related to issue #303 :

I need to use the current users Windows Identity for service authourisation implementation. Our old SOAP service binding is the message wsHttpBinding i.e.

SOAP 1.2
```
	<wsHttpBinding>
		<binding name="WindowsIdentityAuthentication">
		  <security mode="Message">
			<message establishSecurityContext="true" />
		  </security>		  
		</binding>
	</wsHttpBinding> 
```
I configure the service as such:

```
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            AutofacContainer = app.ApplicationServices.GetAutofacRoot();
            WSHttpBinding wsBinding = new WSHttpBinding(SecurityMode.Message);

            app.UseServiceModel(builder =>
            {
                builder
                    .AddService<HelloWorldService>()
                    .AddServiceEndpoint<HelloWorldService, IHelloWorld>(wsBinding, "/HelloWorldService.svc")
                    .ConfigureServiceHostBase<HelloWorldService>(serviceHost =>
                    {
                        ServiceAuthorizationBehavior authBehavior = app.ApplicationServices.GetRequiredService<ServiceAuthorizationBehavior>();
                        var authPolicies = new List<IAuthorizationPolicy>
                        {
                            new HelloWorldServiceSecurityPolicy()
                        };
                        var externalAuthPolicies = new ReadOnlyCollection<IAuthorizationPolicy>(authPolicies);
                        authBehavior.ExternalAuthorizationPolicies = externalAuthPolicies;
                    });
            });
        }
```

The code throws a PlatformNotSupportedException exception on bootstrapping (in the MessageSecurityOverHttp code file).

My question is of course: is this in the pipeline to be implemented as I need to have the Windows Identity on the security context during the ServiceAuthorizationBehaviour implementation i.e. I need to get:
```
var identity= operationContext.ServiceSecurityContext.WindowsIdentity

```

As an aside do I also need to set the following:
```
authBehavior.PrincipalPermissionMode = PrincipalPermissionMode.UseWindowsGroups;
```

Thanks in advance.

**Comment by mconnew** [comment]
You are correct about it being related to issue #303. SecurityMode.Message requires the STR Transform which SignedXml doesn't have an implementation of. It's going to be a while before that gets worked on, it's a complicated piece and has security repercussions. If it's done wrong, it's easy to accidentally introduce a security vulnerability.

**Reply by mconnew** [reply]
As to your question about gitter.im vs here, gitter.im is better if there's a back a forth discussion needed, but sometimes it can take me a few days to notice there's something there. Q&A posts drop something into my email inbox which I check regularly so will often get a faster response, and Q&A has the added benefit of being easily searchable by others. Feel free to use either though.

**Reply by DublinDel** [reply]
Thanks Matt.
